### DISCUSSION TITLE
Welcome to OpenPNM Discussions!

### DISCUSSION THREAD

**Discussion by jgostick** [PMEAL/OpenPNM#1807] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we build together 💪.

**Comment by xu-kai-xu** [comment]
Nice. Hope this discussion will help more and more users.

**Comment by fb638** [comment]
Hi,
I am new to OpenPNM and I want to generate a nonuniform porous network with a specific porosity as well as permeability. I have tried to do it with porespy but in this case, pores are not connected. I will be extremely grateful if you help me. 

thanks in advance

**Reply by xu-kai-xu** [reply]
I find it is difficult to generate a PNM with a specific porosity and permeability. PNM is constructed according to some micro-structure parameters. It is the pore and throat size distributions and pore coordination numbers that determine the porosity and permeability. So, I think the reverse procedure is difficult. I tried to generate PNMs with porosity in the range of 10% - 15%, by repeating generating PNMs and choosing those meeting the conditions. This is the only way I can think about to generate specific porosity. As for permeability, I don't know how to do that.

**Reply by fb638** [reply]
Hi xu-kai-xu,

Thank you for your reply. I was wondering if you could tell me how you generated a porous domain with connected throats and pores area. Please feel free to send any websites or papers. 

Thank you so much.

**Reply by jgostick** [reply]
This question should probably have been written as it's own Q&A, to keep the discussion separated.  I will open up another discussion topic on this issue, and put my answer there.

**Comment by pashaLyb** [comment]
Dear all,

I'm new to OpenPNM and I would like to test it for PTL/GDL characterization. Usually one will use µCT scan to build the network. I'm wondering if it is method to import a PTL structure defined by CAD geometry or STL file? This is interesting for specific kind of the PTL/GDL that have a relatively coarse deterministic structure like metal net.

Thank You

**Reply by jgostick** [reply]
Hi @pashaLyb 
We currently do not have that, only the other way around (export to salome, comsol, stl, etc).  There are two things you're actually asking about here though.  One is to generate a voxel image out of a CAD file so that a network can be extracted. Two is importing a network that is defined by sticks and balls in CAD.  Both are feasible but I've never 'looked' inside a cad file to see how the info is stored.  Would you be able to share one?  Also, feel free to start a fresh 'discusssion' for this subject so we can have the thread in one place.

**Reply by pashaLyb** [reply]
started new discussion.. thanx

**Comment by jluleijiang** [comment]
Dear all,
I'm a new to OpenPNM. Why is there no corresponding content in the model reference? I want to learn more basic knowledge.
thanks a lot.

**Reply by jgostick** [reply]
Can you be more specific?  Making documentation is time consuming and tends to get pushed to the back burner.  We do have a suite of [examples here](https://github.com/PMEAL/OpenPNM/tree/dev/examples).

**Reply by jluleijiang** [reply]
Thank you for your reply. When I imitate examples, I always make mistakes when running, like "TypeError: get_ sections() missing 1 required positional argument: 's'"。 It should be said that the example is not wrong. I thought it was a conflict between my computer and the openpnm version. I replaced different openpnm versions, and the problems were different. So I want to know the basic content. Trouble comes from a completely new.

Thank you so much.

**Reply by jgostick** [reply]
That particular problem is because one of your dependencies is out of data (docrep in this case).  Try uninstalling and reinstalling.  This reinstall should ensure that any dependencies are up to date.  

```
C:\ pip uninstall openpnm
C:\ pip install openpnm
```

Or just update docrep with ``pip install --upgrade docrep``
